server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java

```diff
@@ -361,20 +346,7 @@ public List<RecordResult> query(
                 return 0;
             });
         }
-
-        return results.subList(start, end).stream().map(record -> {
```

Comment:
What are the benefits of moving (start, end) logic outside?

Reply:
In order to support versions, DataStore.query should be able to merge multiple tables just like scan. start & limit only apply to the merged result so that corresponding parameters in MemoryTable.query become meaningless.

Reply:
OK, I get it.